File: frontend/src/Domain/Dtos/CreateUserDto.ts
import AddressDto from '@/domain/dtos/AddressDto.ts';
import EntityBase from '@/domain/EntityBase.ts';

interface CreateUserDto extends EntityBase {
  login: string;
  firstName: string;
  lastName: string;
  role: string;
  phone: string;
  password: string;
  birthDate: string;
  address: AddressDto;
  schoolId?: number;
}

export default CreateUserDto;

File: Application/Dto/Users/CreateUserDto.cs
using Application.Dto.School;
using Domain.Entities_v2.Types;

namespace Application.Dto.Users;

public class CreateUserDto: IBaseDto
{
    public int Id { get; set; }
    public DateTime Created { get; set; }
    public DateTime Updated { get; set; }
    public Status Status { get; set; }
    
    public string Pesel { get; set; } = "";
    public string Login { get; set; } = "";
    public string FirstName { get; set; } = "";
    public string LastName { get; set; } = "";
    public string Role { get; set; } = "";
    public string Phone { get; set; } = "";
    public DateTime BirthDate { get; set; }
    public string Password { get; set; } = "";
    public AddressDto Address { get; set; } = null!;
    public int SchoolId { get; set; }
}